Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h

```diff
@@ -67,6 +67,26 @@ class WebChromeClient final : public WebCore::ChromeClient {
     void relayAccessibilityNotification(String&&, RetainPtr<NSData>&&) const final;
 #endif
 
+    bool isHardwareKeyboardAttached() const override { return m_isAttached; }
+
+    void startHardwareKeyboardAttachmentNotifications(HardwareKeyboardAttachmentCallback&& cb) override
+    {
+        if ((m_callback = WTFMove(cb)))
+            m_callback(m_isAttached);
```

Comment:
Since I'm suggesting to get rid of `m_isAttached` and just reading the value from `WebPage`, just delegate logic to the getter:


```suggestion
            m_callback(isHardwareKeyboardAttached());
```